edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/EdgeAgentConnection.cs

```diff
@@ -368,6 +380,68 @@ Task UpdateDeploymentConfig(TwinCollection desiredProperties)
         async Task<bool> WaitForDeviceClientInitialization() =>
             await Task.WhenAny(this.initTask, Task.Delay(DeviceClientInitializationWaitTime)) == this.initTask;
 
+        private static bool CheckIfTwinPropertiesAreSigned(TwinCollection twinDesiredProperties)
+        {
+            return JObject.Parse(twinDesiredProperties.ToString())["integrity"] != null;
+        }
+
+        public static bool LogTwinSignatureEventsAndReportIfFailed(TwinCollection twinDesiredProperties)
+        {
+            if (!CheckIfTwinPropertiesAreSigned(twinDesiredProperties))
```

Comment:
Instead of dealing with the json blob directly, why don't you deserialize first... will make this code much cleaner...